### PR TITLE
Stop AsyncLoader passing null on cancellation

### PR DESCRIPTION
**Pull request by drewnoakes**
## Proposed changes

Stop AsyncLoader passing null on cancellation. This strange pattern was identified during recent annotation work. It isn't particularly useful to be called back with a default value when the original load action was cancelled.

## Test methodology 

- Manual testing.

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

File: GitCommands/AsyncLoader.cs [hunk 1]
```diff
@@ -73,13 +73,11 @@ public async Task LoadAsync(Action<CancellationToken> loadContent, Action onLoad
                 // Bail early if cancelled, returning default value for type
                 if (token.IsCancellationRequested)
                 {
-                    result = default;
-                }
-                else
-                {
-                    // Load content
-                    result = loadContent(token);
+                    return default;
```
Comment (drewnoakes) [inline GitCommands/AsyncLoader.cs:76]: This is the key change. Now, if cancellation is requested, we just return a default value here. Previously we would still return a default value, but call `onLoaded` as well with that default value.

Comment (gerhardol) [reply GitCommands/AsyncLoader.cs:76]: But this changes the behavior, previously the callers could detect the cancel scenario. In FileViewer for instance, no text would be shown, now an empty string will be displayed.
I believe the change is fine anyway.

**Review by gerhardol — COMMENTED** [review]
+1
Default behavior is slightly changed, probably  not an issue

**Comment by drewnoakes** [comment]
> Default behavior is slightly changed, probably not an issue

Yes, in the case of an operation being cancelled, slightly less work would be performed. I made a review of usages to try and detect any important behaviour changes but didn't find any.

**Review by RussKie — COMMENTED** [review]
LGTM

File: GitCommands/AsyncLoader.cs (hunk 1 above)
Comment (drewnoakes) [reply GitCommands/AsyncLoader.cs:76]: I cannot see the behaviour change in FileViewer. Can you help me to see it? All the callbacks I see performed a no-op when passed null.

**Comment by vbjay** [comment]
The idea is we should be able to walk the token anywhere and use that to
find out if it is canceled.

**Comment by drewnoakes** [comment]
@vbjay that's what happens currently. There's no change in that regard within this PR.

**Pull request merged**